app/controllers/api/v1/registrant/auth_controller.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails5_api_controller_backport'
+require 'auth_token/auth_token_creator'
+
+module Api
+  module V1
+    module Registrant
+      class AuthController < ActionController::API
+        before_action :check_ip_whitelist
+
+        rescue_from(ActionController::ParameterMissing) do |parameter_missing_exception|
+          error = {}
+          error[parameter_missing_exception.param] = ['parameter is required']
+          response = { errors: [error] }
+          render json: response, status: :unprocessable_entity
+        end
+
+        def eid
+          user = RegistrantUser.find_or_create_by_api_data(eid_params)
+          token = create_token(user)
+
+          if token
+            render json: token
+          else
+            render json: { error: 'Cannot create generate session token' }
+          end
+        end
+
+        private
+
+        def eid_params
+          required_params = %i[ident first_name last_name]
+          required_params.each_with_object(params) do |key, obj|
+            obj.require(key)
+          end
+
+          params.permit(required_params)
+        end
+
+        def create_token(user)
+          token_creator = AuthTokenCreator.create_with_defaults(user)
+          hash = token_creator.token_in_hash
+          hash
+        end
+
+        def check_ip_whitelist
+          allowed_ips = ENV['registrant_api_auth_allowed_ips'].to_s.split(',').map(&:strip)
+          return if allowed_ips.include?(request.ip) || Rails.env.development?
```

Comment:
Any chance to remove hardcoded env check and make it configurable?

Reply:
I was considering using an environment variable, but decided against it.

If we use `registrant_api-auth_allowed_ips` (leave blank to allow any IP), that's a bit unclear and leaves a lot of room for configuration error. Figaro does not allow for expressing more complex data structures.

If we create a new variable that's used exactly for allowing all requests to create tokens, that's making it too complicated for ourselves, as the only use of this thing ever is development environment.

It might be worth a try to use a regex to instead of a list, I'd need to investigate and test that.

Reply:
I did some additional research on this, I thought we could use `ipaddr` library to add some more interesting parsing, such as suport for subnets and IPv6, but it's extremely slow for anything with `/16` or smaller subnet mask. 

/16:
```ruby
Benchmark.bm do |x|
  x.report { ip_addr =IPAddr.new('192.168.1.1/16'); ip_addr.to_range.include?('182.0.0.1') }
  x.report { ip_addr =IPAddr.new('192.168.1.1/16'); ip_addr.to_range.include?('192.168.0.1') }
  x.report { ip_addr =IPAddr.new('192.168.1.1/16'); ip_addr.to_range.include?('0.1.0.1') }
  x.report { ip_addr =IPAddr.new('192.168.1.1/16'); ip_addr.to_range.include?('0.1.2.1') }
end

       user     system      total        real
   0.420000   0.000000   0.420000 (  0.417766)
   0.000000   0.000000   0.000000 (  0.000038)
   0.420000   0.000000   0.420000 (  0.423992)
   0.430000   0.000000   0.430000 (  0.428151)
```

/8:
```ruby
Benchmark.bm do |x|
  x.report { ip_addr =IPAddr.new('192.168.1.1/8'); ip_addr.to_range.include?('0.1.2.1') }
  x.report { ip_addr =IPAddr.new('192.168.1.1/8'); ip_addr.to_range.include?('192.1.2.1') }
end
       user     system      total        real
 100.980000   0.000000 100.980000 (101.159714)
   0.420000   0.000000   0.420000 (  0.419837)
```

The hard-coded environment check is there to stay.